python/lsst/daf/butler/registries/oracleRegistry.py

```diff
@@ -107,6 +120,28 @@ def _oracleExecute(connection, cursor, statement, parameters, context, executema
                     # single quote, optional close double quote
                     statement = re.sub(f"\"?'?{name}\\b'?\"?", lambda x: _ignoreQuote(name, x), statement)
             return statement, parameters
-        engine = create_engine(self.config["db"], pool_size=1)
+
+        def _connCreator(**kw):
+            """ This function sets some Oracle values for each new pool
+            connection (e.g., module, schema).  Add connect listener to
+            Pool even if pool_size=1 for automatic reconnects.
+            """
+            # get executable name to save as DBMS_APPLICATION_INFO module
+            try:
+                modname = __main__.__file__
+            except AttributeError:
+                modname = "Unknown"
+            modname = os.path.basename(modname)[:48]  # column limited to 48 chars
```

Comment:
Do you need to drop path information?  Can't you store the final 48 characters regardless so that you get some idea of where the executable came from?
```
modname = modname[-48:]
```
?

Reply:
Mostly it was just an assumption that this extra information could be determined via other means (like software provenance) and didn't want to require use of wildcards in searches.
   
Example final 48 characters: 
- _butler/17.0.1-12-gd4f779f/bin/makeButlerRepo.py
- ipe_tasks/17.0.1-8-g40ca8d54/bin/ingestImages.py
- x64/ctrl_mpexec/17.0.1-4-g65ad0af+2/bin/pipetask

I checked with the DB admins.   There are no technical reasons to not store full 48 characters (space not an issue, no indexing to be affected by use of a wildcard for beginning of string, etc).   Will change code to always store last 48 characters.